inst/include/CytoML/flowJoWorkspace.hpp

```diff
@@ -403,63 +406,96 @@ class flowJoWorkspace:public workspace{
 	 * cytoframe is preloaded with header-only.
 	 */
-	bool search_for_fcs(const string & data_dir, const int sample_id, const string & sample_name, const string & ws_key_seq, const ParseWorkspaceParameters & config, shared_ptr<MemCytoFrame> &fr)
+	bool search_for_fcs(const string & data_dir, const int sample_id, const string & sample_name, string & uid, const int & total_event_count, const string & ws_key_seq, const ParseWorkspaceParameters & config, shared_ptr<MemCytoFrame> &fr)
 	{
 		FCS_READ_PARAM fcs_read_param = config.fcs_read_param;
 		fcs_read_param.header.is_fix_slash_in_channel_name = is_fix_slash_in_channel_name();
 		if(g_loglevel>=GATING_HIERARCHY_LEVEL)
-			COUT<<endl<<"searching for FCS for sample: " + sample_name <<endl;
+			COUT<<endl<<"Searching for FCS for sample: " + sample_name <<endl;
 		//try to search by file name first
 		vector<string> all_file_paths = list_files(data_dir, config.fcs_file_extension);
 		vector<string> file_paths;
-		for(auto i : all_file_paths)
-		{
+		vector<string> file_paths_clipped;
+		bool isfound = false;
+		
+		// Gather files that match sample_name
+		for(auto i : all_file_paths){
 			if(path_base_name(i) == sample_name)
 				file_paths.push_back(i);
 		}
-		bool isfound = false;
-		if(file_paths.size() == 0)
-		{
-			//search by keyword by traversing the data dir recursively until the target is found
-			for(const string & file_path : all_file_paths)
-			{
-
+		
+		// No matches found from simple filename check. Check $FIL keyword for prospective match.
+		if(file_paths.size() == 0){
+			PRINT("No FCS file found with filename matching sample name " + sample_name + ". Checking for files with $FIL keyword matching sample name.");
+			for(const string & file_path : all_file_paths){
 				fr.reset(new MemCytoFrame(file_path, fcs_read_param));
 				fr->read_fcs_header();
-				if(fr->get_keyword("$FIL") == sample_name &&
-						ws_key_seq == concatenate_keywords(fr->get_keywords(), config.keywords_for_uid, config.keywords_for_uid_sampleID, sample_id))
-				{
-
-					isfound = true;
-					break;//to avoid scanning entire folder, terminate the search immediately on the first hit assuming the uid is unique
-
-				}
+				// Just gather those with $FIL matching sample name. If necessary, keyword check will occur in subsequent logic.
+				if(fr->get_keyword("$FIL") == sample_name)
+					file_paths.push_back(file_path);
 			}
-
-
 		}
-		else
-		{//further verify/pruning the matched files by looking at the keywords
-
-			for(const string & file_path : file_paths)
-			{
+		
+		if(file_paths.size() == 0){
+			PRINT("No FCS file found for sample name " + sample_name + ".");
+		}else if( file_paths.size() == 1 ){ // Unambiguous match
+			fr.reset(new MemCytoFrame(file_paths[0], fcs_read_param));
+			fr->read_fcs_header();
+			if(std::stoi(fr->get_keyword("$TOT")) != total_event_count){
```

Comment:
`std` namespace prefixes are unnecessary. I'll clip them out in a cleanup commit later.